crates/cli/cli/src/commands/query/client/status.rs

```diff
@@ -0,0 +1,144 @@
+    HasConsensusStateFields, HasConsensusStateType,
+};
+use oneline_eyre::eyre::Context;
+use serde::Serialize;
+
+use hermes_relayer_components::chain::traits::queries::chain_status::CanQueryChainStatus;
+use hermes_relayer_components::chain::traits::queries::consensus_state::CanQueryConsensusState;
+
+use hermes_cli_framework::command::CommandRunner;
+use hermes_cli_framework::output::Output;
+use hermes_cosmos_relayer::types::error::BaseError;
+use hermes_relayer_components::birelay::traits::two_way::HasTwoWayRelayTypes;
+use hermes_relayer_components::build::traits::components::chain_builder::CanBuildChain;
+use hermes_relayer_components::build::traits::target::chain::ChainATarget;
+use hermes_relayer_components::chain::traits::queries::client_state::{
+    CanQueryClientState, CanQueryClientStateWithHeight,
+};
+use hermes_relayer_components::chain::traits::types::client_state::{
+    HasClientStateFields, HasClientStateType,
+};
+use hermes_relayer_components::chain::traits::types::ibc::HasIbcChainTypes;
+use ibc_relayer_types::core::ics02_client::height::Height;
+use ibc_relayer_types::core::ics24_host::identifier::{ChainId, ClientId};
+
+use crate::Result;
+
+#[derive(Debug, clap::Parser)]
+pub struct QueryClientStatus {
+    /// Identifier of the host chain
+    #[clap(
+        long = "chain",
+        required = true,
+        value_name = "CHAIN_ID",
+        help_heading = "REQUIRED"
+    )]
+    chain_id: ChainId,
+
+    /// Identifier of the client on the host chain
+    #[clap(
+        long = "client",
+        required = true,
+        value_name = "CLIENT_ID",
+        help_heading = "REQUIRED"
+    )]
+    client_id: ClientId,
+}
+
+impl<Build, Chain, Counterparty> CommandRunner<Build> for QueryClientStatus
+where
+    Build: CanBuildChain<ChainATarget>,
+    Build::BiRelay: HasTwoWayRelayTypes<ChainA = Chain, ChainB = Counterparty>,
+    Chain: HasIbcChainTypes<Counterparty, ChainId = ChainId, ClientId = ClientId, Height = Height>
+        + CanQueryClientState<Counterparty>
+        + CanQueryClientStateWithHeight<Counterparty>
+        + CanQueryChainStatus
+        + CanQueryConsensusState<Counterparty>,
+    Counterparty: HasIbcChainTypes<Chain>
+        + HasClientStateType<Chain>
+        + HasClientStateFields<Chain>
+        + HasConsensusStateType<Chain>
+        + HasConsensusStateFields<Chain>,
+    Chain::Error: From<BaseError> + StdError,
+    Build::Error: From<BaseError> + StdError,
+    Counterparty::ClientState: Serialize,
+{
+    async fn run(&self, builder: &Build) -> Result<Output> {
+        let chain = builder.build_chain(ChainATarget, &self.chain_id).await?;
+        let client_status = query_client_status(&chain, &self.client_id).await?;
+
+        Ok(Output::success(client_status))
+    }
+}
+
+#[derive(Debug, Serialize)]
+enum Status {
+    Frozen,
+    Expired,
+    Active,
+}
+
+async fn query_client_status<Chain, Counterparty>(
```

Comment:
@soareschen Shall I move this into a `CanQueryClientStatus` trait/impl or do we leave it as is for now?

Reply:
I am giving it a try